### ISSUE TITLE
Find unlinked files should propose to link to existing entry (if it exists)

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#7288]
Source: https://github.com/JabRef/jabref/pull/7172#issuecomment-745293123

Instead of only suggesting importing the pdf as a new entry, JabRef should also suggest to link the file to a fitting existing entry (by adding a button in the file tree next to the file for which an entry might already exist).

**Comment by ilippert** [comment]
I think I offer a relevant and related way of using JabRef: I have a folder in which all the linked files of JabRef are located. 
I have also a different folder in which I collect pdfs that are not yet linked at all to JabRef. Once I link them I also move them to JabRef.

If possible, it would be amazing if JabRef could search one/a number of specified other folders also for combinations of author names and years or titles.

**Comment by ilippert** [comment]
still is a good idea

**Comment by k3KAW8Pnf7mkmdSMPHz27** [comment]
@ilippert, is the issue with

1. "Autolink files" with "Use regular expression search": Files aren't automatically linked?
2. The menu "Lookup" with "Search for unlinked local files": Doesn't deal with duplicated bibentries?

**Comment by ilippert** [comment]
Well, my proposal was to think more broadly: one dimension mentioned in https://github.com/JabRef/jabref/issues/7288#issue-778185582 is that JabRef could search the "Main file directory" for pdfs and also offer to link these to already existing entries in the library. Another dimension is to add more folders, beyond the "Main file directory" for searching.

**Comment by koppor** [comment]
For the first dimension (which I meant here): We could use the duplication check algorithm "magically" during the synchronization. - Detail question @ilippert: Would you like to have more control what JabRef does (e.g., by deciding each file one-by-one) or trust the "JabRef magic" and fix minor issues afterwards?

For the second dimension: Files in these folders are linked in absolute paths. Would that be OK for you @ilippert.

**Comment by ilippert** [comment]
> For the first dimension (which I meant here): We could use the duplication check algorithm "magically" during the synchronization. - Detail question @ilippert: Would you like to have more control what JabRef does (e.g., by deciding each file one-by-one) or trust the "JabRef magic" and fix minor issues afterwards?
I would love a "wizard" that asks me for each case, what to do about the file. As in: `I found file X in folder Y. It does not belong to an entry in your library yet. But entry Z (give me a hover tip on the entry's content) seems to match file X. Check this checkbox, if you want these to be linked (or moved)`

I could imagine a table-like layout:
column 1: file x@folder y. 
column 2: candidate entry
column 3: action (link, move, rename or more?)
below the table a line asking to apply the changes.
 
> For the second dimension: Files in these folders are linked in absolute paths. Would that be OK for you @ilippert.
absolute paths are perfectly fine. (And if I ever needed relative paths, I would ref to a symbolic link that itself is relative).

**Comment by koppor** [comment]
I refined that issue at https://github.com/JabRef/jabref/issues/9798. There, I described a) an automatic way for "Autoset file links" - and refined the "Search for unlinked local files" dialog.

The issue with "Support multiple file directories" will be refined. We need to think about a) the UI and b) the logic. (I very well understand that issue. I treated it differently somehow: https://github.com/JabRef/jabref/issues/160)
